pages/user/RegistrationHandler.inc.php

```diff
@@ -78,10 +78,19 @@ function registerUser($args, &$request) {
 
 		if ($regForm->validate()) {
 			$regForm->execute();
-			if (Config::getVar('email', 'require_validation')) {
-				// Send them home; they need to deal with the
-				// registration email.
-				$request->redirect(null, 'index');
+			if (!Validation::isLoggedIn()) {
+				if (Config::getVar('email', 'require_validation')) {
+					// Send them home; they need to deal with the
+					// registration email.
+					$request->redirect(null, 'index');
```

Comment:
The code below definitely executes but I'll remove the line because the redirect isn't what should be happening now.